sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2095,7 +2095,9 @@ case class ArrayPosition(left: Expression, right: Expression)
 case class ElementAt(
     left: Expression,
     right: Expression,
-    failOnError: Boolean = SQLConf.get.ansiEnabled)
+    failOnError: Boolean = SQLConf.get.ansiEnabled,
+    // The value to return if index is out of bound
+    defaultValueOutOfBound: Any = null)
```

Comment:
I'm a little worried about this change, as it's tricky to support default value here. e.g. what if the element type is array/map/struct? what value we should put here? and how to use it in codegen?

Instead, I think we need to use `Expression` to represent the default value, as this is the "language" catalyst speaks. We can require it to be foldable to simplify the implementation.

Reply:
Is there an example of how to create a Literal("") for Expression type of `defaultValueOutOfBound`?

I tried `Literal("")`, etc. It always gives me this error: 

```
requirement failed: Literal must have a corresponding value to string, but class String found.
java.lang.IllegalArgumentException: requirement failed: Literal must have a corresponding value to string, but class String found.
```

Reply:
I think existing code base has been using `Literal("")` for empty string as Expression. Not sure why it fails in my case when doing the replacement for `split_part`.

Reply:
I updated my change. Hopefully you could give me some suggestions.